[Bug]: Error in PR-Check (Type-check) failure

### Describe the bug

Run mypy
  mypy
  shell: /bin/bash -e {0}
  env:
    UV_CACHE_DIR: /Users/runner/work/_temp/setup-uv-cache
    pythonLocation: /Users/runner/hostedtoolcache/Python/3.9.13/arm64
    PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.9.13/arm64/lib/pkgconfig
    Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.13/arm64
    Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.13/arm64
    Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.13/arm64
test/oai/test_oai_models.py:37: error: "<typing special form>" has no attribute "model_json_schema"  [attr-defined]
Found 3 errors in 2 files (checked 361 source files)
autogen/tools/experimental/web_search_preview/web_search_preview.py:98: error: Returning Any from function declared to return "Union[str, BaseModel, None]"  [no-any-return]
autogen/tools/experimental/web_search_preview/web_search_preview.py:98: error: "Response" has no attribute "output_text"  [attr-defined]
Error: Process completed with exit code 1.

### Steps to reproduce

_No response_

### Model Used

_No response_

### Expected Behavior

_No response_

### Screenshots and logs

_No response_

### Additional Information

_No response_

--- a/autogen/tools/experimental/web_search_preview/web_search_preview.py
+++ b/autogen/tools/experimental/web_search_preview/web_search_preview.py
@@ -95,7 +95,13 @@ class WebSearchPreviewTool(Tool):
                     input=query,
                     instructions=self.instructions,
                 )
-                return response.output_text
+                text = []
+                for output in response.output:
+                    if output.type == "message":
+                        for content in output.content:
+                            if content.type == "output_text":
+                                text.append(content.text)
+                return "\n".join(text)
 
             else:
                 response = client.responses.parse(

--- a/test/oai/test_oai_models.py
+++ b/test/oai/test_oai_models.py
@@ -18,9 +18,10 @@ from autogen.oai.oai_models import (
 from autogen.oai.oai_models.chat_completion import ChatCompletion as ChatCompletionLocal
 
 with optional_import_block():
-    from openai.types.chat import ChatCompletion, ChatCompletionMessageToolCall
+    from openai.types.chat import ChatCompletion
     from openai.types.chat.chat_completion import Choice
     from openai.types.chat.chat_completion_message import ChatCompletionMessage
+    from openai.types.chat.chat_completion_message_function_tool_call import ChatCompletionMessageFunctionToolCall
     from openai.types.completion_usage import CompletionUsage
 
 
@@ -34,7 +35,8 @@ class TestOAIModels:
 
     def test_chat_completion_message_tool_call_schema(self) -> None:
         assert (
-            ChatCompletionMessageToolCallLocal.model_json_schema() == ChatCompletionMessageToolCall.model_json_schema()
+            ChatCompletionMessageToolCallLocal.model_json_schema()
+            == ChatCompletionMessageFunctionToolCall.model_json_schema()
         )
 
     def test_choice_schema(self) -> None:

